internal/handlers/common/aggregations/group.go

```diff
@@ -205,31 +204,8 @@ type idAccumulator struct {
 //
 //	{$group: {_id: "$v"}} sets the value of $v to _id, Accumulate returns the value found at key `v`.
 //	{$group: {_id: null}} sets `null` to _id, Accumulate returns nil.
-func (a *idAccumulator) Accumulate(ctx context.Context, in []*types.Document) (any, error) {
-	groupKey, ok := a.expression.(string)
-	if !ok {
-		return a.expression, nil
-	}
-
-	if !strings.HasPrefix(groupKey, "$") {
-		return groupKey, nil
-	}
-
-	key := strings.TrimPrefix(groupKey, "$")
-
-	path, err := types.NewPathFromString(key)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
-
-	// use the first element, it was already grouped by the groupKey,
-	// so all `in` documents contain the same v.
-	v, err := in[0].GetByPath(path)
-	if err != nil {
-		return types.Null, nil
-	}
-
-	return v, nil
+func (a *idAccumulator) Accumulate(ctx context.Context, groupID any, in []*types.Document) (any, error) {
+	return groupID, nil
 }
```

Comment:
The code looks a bit confusing to me.

There's an `Accumulator` interface, it's usage is bit unclear, as it seems that all implementing structs doesn't accumulate any data.
After a while I realized it could be representation of MongoDB `$group`'s accumulators, like `$count`.
We have a nice map `accumulators` map that shows every supported accumulator, cool.

Afterwards, there's `idAccumulator`, which also implements the same `Accumulator` interface, but it doesn't exist as accumulator in mongodb docs.
It looks like some internal struct to handle `_id` fields, but it's under the same interface as `$count`, which is MongoDB functionality, right?

I understand logic more or less but at the first glance it was really confusing to understand what's responsible for what.

I assume that I could misunderstood something, so I don't have many ideas how to improve it, just wanted to leave it here, to
maybe consider future refactorings or not :)

Reply:
Great feedback :hugs:  Let's not make `idAcumulator` be part of `Accumulator` implementation!